aten/src/ATen/NestedTensorImpl.cpp

```diff
@@ -85,5 +115,34 @@ const char* NestedTensorImpl::tensorimpl_type_name() const {
   return "NestedTensorImpl";
 }
 
+// implicit batch dimension index offsets, original tensors shapes
+std::tuple<std::vector<int64_t>, std::vector<IntArrayRef>>
+NestedTensorImpl::get_offsets_and_shapes() const {
```

Comment:
I'd prefer if this was two separate functions instead of a method. They're utilities that are useful for a few kernels, but I don't think they should become part of the NestedTensorImpl API.

Reply:
made functions `NestedTensor_get_offsets` and `NestedTensor_get_shapes` in NestedTensorMath.cpp